40.md

```diff
@@ -0,0 +1,53 @@
+NIP-40
+======
+
+Expiration Timestamp
+-----------------------------------
+
+`draft` `optional` `author:0xtlt`
+
+The `expiration` tag enables users to specify a timestamp at UTC-0 at which the message should be considered expired and deleted by relays (or just ignored by clients). This can be useful for making temporary announcements, making limited-time offers, or sending ephemeral messages but should not be seen as a security solution for the content of your messages.
+So be careful when sharing confidential information.
+
+#### Spec
+
+```
+tag: expiration
+options:
+ - [UNIX timestamp in seconds]: required
+```
+
+#### Example
+
+```json
+{
+    "pubkey": "<pub-key>",
+    "created_at": 1000000000,
+    "kind": 1,
+    "tags": [
+      ["expiration", 1600000000]
+    ],
+    "content": "This message will expire at the specified timestamp and be deleted by relays.\n",
+    "id": "<event-id>"
+}
+```
+
+Note: The timestamp should be in the same format as the created_at timestamp and should be interpreted as the time at which the message should be deleted by relays.
+
+Client Behavior
+---------------
+
+Clients SHOULD use the `supported_nips` field to learn if a relay supports this NIP. Clients SHOULD NOT send expiration events to relays that do not support this NIP.
+
+Clients MUST ignore events that have expired.
+
```

Comment:
As the expiration time must be at UTC-0, the local times is not a problem here

Reply:
@x1ddos said
> As the expiration time must be at UTC-0, the local times is not a problem here

There is no such thing as UTC-0 in a decentralized network.

Services that rely on an accurate time, like 2FA or TOTP, can run into issues because of time drift:

* https://blog.gaborszathmari.me/tricking-google-authenticator-totp-with-ntp/
* https://www.reddit.com/r/synology/comments/f9lu53/2fa_and_ntp_woes/
* https://forum.matomo.org/t/2fa-authentication-suddenly-fails/45829

Basing message deletion on that might be asking for trouble :)